File: #27_Remove_Element/Solution.java
public class Solution {
    public int removeElement(int[] nums, int val) {
    	if (nums == null || nums.length == 0) return 0;
    	int len = nums.length;
    	int count = 0;
    	int i = 0, j = len - 1;
    	while (i <= j) {
    		if (nums[i] == val) {
    			if (nums[j] == val) {
    				j--; count++;
    				continue;
    			}
    			nums[i] = nums[j];
    			nums[j] = val;
    			j--; count++;
    		}
    		i++;
    	}
    	return len - count;
    }

    public static void main(String[] args) {
    	Solution sol = new Solution();
    	int[] t = {1, 2, 2, 3};
    	System.out.println(sol.removeElement(t, 2));
    	for (int it : t) {
    		System.out.print(it + " ");
    	}
    	System.out.println();
    }
}

File: #27_Remove_Element/Solution.py
class Solution:
	# @param {integer[]} nums
	# @param {integer} val
	# @return {integer}
	def removeElement(self, nums, val):
		if not nums:
			return 0
		s, f = 0, 0
		while f < len(nums):
			if nums[f] != val:
				nums[s] = nums[f]
				s += 1
			f += 1
		return s